substrate/frame/revive/src/precompiles/custom/xcm.rs

```diff
@@ -0,0 +1,114 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::Origin;
+use crate::{
+	precompiles::{AddressMatcher, Error, Ext, ExtWithInfo, Precompile},
+	Config,
+};
+use alloc::vec::Vec;
+use alloy_core::sol;
+use codec::{DecodeAll, Encode};
+use core::{marker::PhantomData, num::NonZero};
+use xcm_builder::{ExecuteController, ExecuteControllerWeightInfo, SendController};
+pub use IXcm::IXcmCalls;
+
+pub struct XCMPrecompile<T>(PhantomData<T>);
```

Comment:
I see `IXcm` / `IXcmCalls` ..., so maybe also:
```suggestion
pub struct XcmPrecompile<T>(PhantomData<T>);
```

Reply:
Yup fixed

Reply:
@bkontur @franciscoaguirre do yall have a better naming convention for the precompile? I figured Xcm+Precompile is sufficient given how many other things are named xcm?